metrics/sources/kubelet/summary.go

```
@@ -0,0 +1,342 @@
+	this.addIntMetric(metrics, &MetricMemoryPageFaults, memory.PageFaults, startTime)
+	this.addIntMetric(metrics, &MetricMemoryMajorPageFaults, memory.MajorPageFaults, startTime)
+}
+
+func (this *summaryMetricsSource) decodeNetworkStats(metrics *MetricSet, network *stats.NetworkStats, startTime time.Time) {
+	if network == nil {
+		return
+	}
+
+	this.addIntMetric(metrics, &MetricNetworkRx, network.RxBytes, startTime)
+	this.addIntMetric(metrics, &MetricNetworkRxErrors, network.RxErrors, startTime)
+	this.addIntMetric(metrics, &MetricNetworkTx, network.TxBytes, startTime)
+	this.addIntMetric(metrics, &MetricNetworkTxErrors, network.TxErrors, startTime)
+}
+
+func (this *summaryMetricsSource) decodeFsStats(metrics *MetricSet, fsKey string, fs *stats.FsStats) {
+	if fs == nil {
+		return
+	}
+
+	fsLabels := map[string]string{LabelResourceID.Key: fsKey}
+	if fs.UsedBytes != nil {
+		usage := LabeledMetric{
+			Name:       MetricFilesystemUsage.Name,
+			Labels:     fsLabels,
+			ValueType:  ValueInt64,
+			MetricType: MetricFilesystemUsage.Type,
+			IntValue:   int64(*fs.UsedBytes),
+		}
+		metrics.LabeledMetrics = append(metrics.LabeledMetrics, usage)
+	}
+
+	if fs.CapacityBytes != nil {
+		limit := LabeledMetric{
+			Name:       MetricFilesystemLimit.Name,
+			Labels:     fsLabels,
+			ValueType:  ValueInt64,
+			MetricType: MetricFilesystemLimit.Type,
+			IntValue:   int64(*fs.CapacityBytes),
+		}
+		metrics.LabeledMetrics = append(metrics.LabeledMetrics, limit)
+	}
+}
+
+func (this *summaryMetricsSource) decodeUserDefinedMetrics(metrics *MetricSet, udm []stats.UserDefinedMetric) {
+	for _, metric := range udm {
+		mv := MetricValue{}
+		switch metric.Type {
+		case stats.MetricGauge:
+			mv.MetricType = MetricGauge
+		case stats.MetricCumulative:
+			mv.MetricType = MetricCumulative
+		default:
+			glog.V(4).Infof("Skipping %s: unknown custom metric type: %v", metric.Name, metric.Type)
```

Will rate or delta metrics be silently dropped?